main/main.c

```diff
@@ -142,10 +196,29 @@ run_signal_handler(void)
 int
 main(int argc, char **argv)
 {
+	struct args args = {
+		.lua_base_dir = NULL,
+		.gatekeeper_config_file = NULL,
+	};
+
 	int ret = rte_eal_init(argc, argv);
 	if (ret < 0)
 		rte_exit(EXIT_FAILURE, "Error with EAL initialization\n");
 
+	argc -= ret;
+	argv += ret;
+
+	/* Parse Gatekeeper arguments (after the EAL ones). */
+	ret = argp_parse(&argp, argc, argv, 0, NULL, &args);
+	if (ret != 0)
+		rte_exit(EXIT_FAILURE, "Invalid Gatekeeper parameters\n");
+
+	if (args.lua_base_dir)
+		lua_base_dir = rte_strdup("lua_base_dir", args.lua_base_dir);
+	if (args.gatekeeper_config_file)
+		gatekeeper_config_file = rte_strdup("gatekeeper_config_file",
```

Comment:
Given that `args.lua_base_dir` and `args.gatekeeper_config_file` points to memory that will only go away once Gatekeeper finishes, we can directly assign them to the global variables.